sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -195,84 +229,166 @@ public void getTempFilePrefix(ProcessContext c) {
                             tempLocation);
-                        c.output(tempLocation);
+                          c.output(tempLocation);
                       }
                     }))
             .apply("TempFilePrefixView", View.<String>asSingleton());
 
-    PCollection<KV<DestinationT, TableRow>> inputInGlobalWindow =
-        input.apply(
-            "rewindowIntoGlobal",
-            Window.<KV<DestinationT, TableRow>>into(new GlobalWindows())
-                .triggering(DefaultTrigger.of())
-                .discardingFiredPanes());
-    PCollectionView<Map<DestinationT, String>> schemasView =
-        inputInGlobalWindow.apply(new CalculateSchemas<>(dynamicDestinations));
-
-    TupleTag<WriteBundlesToFiles.Result<DestinationT>> writtenFilesTag =
-        new TupleTag<WriteBundlesToFiles.Result<DestinationT>>("writtenFiles"){};
-    TupleTag<KV<ShardedKey<DestinationT>, TableRow>> unwrittedRecordsTag =
-        new TupleTag<KV<ShardedKey<DestinationT>, TableRow>>("unwrittenRecords") {};
-    PCollectionTuple writeBundlesTuple = inputInGlobalWindow
-            .apply("WriteBundlesToFiles",
-                ParDo.of(new WriteBundlesToFiles<>(tempFilePrefix, unwrittedRecordsTag,
-                    maxNumWritersPerBundle, maxFileSize))
-                    .withSideInputs(tempFilePrefix)
-                    .withOutputTags(writtenFilesTag, TupleTagList.of(unwrittedRecordsTag)));
-    PCollection<WriteBundlesToFiles.Result<DestinationT>> writtenFiles =
-        writeBundlesTuple.get(writtenFilesTag)
-        .setCoder(WriteBundlesToFiles.ResultCoder.of(destinationCoder));
-
-    // If the bundles contain too many output tables to be written inline to files (due to memory
-    // limits), any unwritten records will be spilled to the unwrittenRecordsTag PCollection.
-    // Group these records by key, and write the files after grouping. Since the record is grouped
-    // by key, we can ensure that only one file is open at a time in each bundle.
-    PCollection<WriteBundlesToFiles.Result<DestinationT>> writtenFilesGrouped =
-        writeBundlesTuple
-            .get(unwrittedRecordsTag)
-            .setCoder(KvCoder.of(ShardedKeyCoder.of(destinationCoder), TableRowJsonCoder.of()))
-            .apply(GroupByKey.<ShardedKey<DestinationT>, TableRow>create())
-            .apply(
-                ParDo.of(new WriteGroupedRecordsToFiles<DestinationT>(tempFilePrefix, maxFileSize))
-                    .withSideInputs(tempFilePrefix))
-            .setCoder(WriteBundlesToFiles.ResultCoder.of(destinationCoder));
-
     // PCollection of filename, file byte size, and table destination.
-    PCollection<WriteBundlesToFiles.Result<DestinationT>> results =
-        PCollectionList.of(writtenFiles).and(writtenFilesGrouped)
-        .apply(Flatten.<Result<DestinationT>>pCollections());
+    PCollection<WriteBundlesToFiles.Result<DestinationT>> results;
+    if (numFileShards == 0) {
```

Comment:
Basically if triggeringFrequency == null, right? I think that condition would be more clear here. And add a comment about what it means.